Packs/AzureKubernetesServices/Integrations/AzureKubernetesServices/AzureKubernetesServices.py

```diff
@@ -0,0 +1,194 @@
+            scope='https://management.azure.com/user_impersonation offline_access user.read',
+        )
+        self.subscription_id = subscription_id
+        self.resource_group_name = resource_group_name
+
+    @logger
+    def clusters_list_request(self) -> Dict:
+        return self.ms_client.http_request(
+            method='GET',
+            url_suffix='providers/Microsoft.ContainerService/managedClusters',
+            params={
+                'api-version': API_VERSION,
+            },
+        )
+
+    @logger
+    def cluster_get(self, resource_name: str) -> Dict:
+        return self.ms_client.http_request(
+            method='GET',
+            url_suffix=f'resourceGroups/{self.resource_group_name}/providers/Microsoft.ContainerService/managedClusters'
+                       f'/{resource_name}',
+            params={
+                'api-version': API_VERSION,
+            },
+        )
+
+    @logger
+    def cluster_addon_update(self,
+                             resource_name: str,
+                             location: str,
+                             http_application_routing_enabled: Optional[bool] = None,
+                             monitoring_agent_enabled: Optional[bool] = None,
+                             monitoring_resource_name: Optional[str] = None,
+                             ) -> Dict:
+        addon_profiles: Dict[str, Any] = {}
+        if http_application_routing_enabled is not None:
+            addon_profiles['httpApplicationRouting'] = {'enabled': http_application_routing_enabled}
+        if monitoring_agent_enabled is not None:
+            if monitoring_resource_name:
+                workspace_resource_id = f'/subscriptions/{self.subscription_id}/resourceGroups/' \
+                                        f'DefaultResourceGroup-WUS/providers/Microsoft.OperationalInsights/workspaces' \
+                                        f'/{monitoring_resource_name}'
+            else:
+                cluster = self.cluster_get(resource_name)
+                workspace_resource_id = cluster.get('properties', {}).get('addonProfiles', {}).get('omsagent', {})\
+                    .get('config', {}).get('logAnalyticsWorkspaceResourceID')
+            addon_profiles['omsagent'] = {
+                'enabled': monitoring_agent_enabled,
+                'config': {'logAnalyticsWorkspaceResourceID': workspace_resource_id},
+            }
+        return self.ms_client.http_request(
+            'PUT',
+            url_suffix=f'resourceGroups/{self.resource_group_name}/providers/Microsoft.ContainerService/managedClusters'
+                       f'/{resource_name}',
+            params={
+                'api-version': API_VERSION,
+            },
+            json_data={
+                'location': location,
+                'properties': {
+                    'addonProfiles': addon_profiles,
+                }
+            },
+            timeout=30,
```

Comment:
why so long?

Reply:
from some tests, it takes about 10-15 seconds, so wanted to stay on the safe side
do you think its too much?